plugins/modules/ec2_vpc_route_table.py

```diff
@@ -510,29 +538,63 @@ def ensure_subnet_associations(connection=None, module=None, route_table=None, s
         new_association_ids.append(result['association_id'])
 
     if purge_subnets:
-        to_delete = [a_id for a_id in current_association_ids
-                     if a_id not in new_association_ids]
+        to_delete = [association_id for association_id in current_association_ids
+                     if association_id not in new_association_ids]
 
-        for a_id in to_delete:
+        for association_id in to_delete:
             changed = True
             if not check_mode:
                 try:
-                    connection.disassociate_route_table(aws_retry=True, AssociationId=a_id)
+                    connection.disassociate_route_table(aws_retry=True, AssociationId=association_id)
                 except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
                     module.fail_json_aws(e, msg="Couldn't disassociate subnet from route table")
 
     return {'changed': changed}
 
 
+def ensure_gateway_association(connection=None, module=None, route_table=None, gateway_id=None,
+                               check_mode=None):
+    changed = False
+    filters = ansible_dict_to_boto3_filter_list({'association.gateway-id': gateway_id, 'vpc-id': route_table['VpcId']})
+    try:
+        route_tables = describe_route_tables_with_backoff(connection, Filters=filters)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't get route tables")
+    for table in route_tables:
+        if table['RouteTableId'] is None:
+            continue
+        for association in table['Associations']:
+            if association['Main']:
+                continue
+            if association['GatewayId'] == gateway_id:
+                if table['RouteTableId'] == route_table['RouteTableId']:
+                    return {'changed': False, 'association_id': association['RouteTableAssociationId']}
+                if check_mode:
+                    return {'changed': True}
+                try:
+                    connection.disassociate_route_table(
+                        aws_retry=True, AssociationId=association['RouteTableAssociationId'])
+                except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                    module.fail_json_aws(e, msg="Couldn't disassociate gateway from route table")
+
+    try:
+        connection.associate_route_table(aws_retry=True,
+                                         RouteTableId=route_table['RouteTableId'],
+                                         GatewayId=gateway_id)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't associate gateway with route table")
+    return {'changed': changed}
```

Comment:
I guess this spoil be enough to handle check_mode in this case.
```suggestion
    if not module.check_mode:
        try:
            connection.associate_route_table(aws_retry=True,
                                             RouteTableId=route_table['RouteTableId'],
                                             GatewayId=gateway_id)
        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
            module.fail_json_aws(e, msg="Couldn't associate gateway with route table")
    return {'changed': changed}
```